include/RAJA/exec-cuda/forall_cuda.hxx

```diff
@@ -150,9 +176,12 @@ __global__ void forall_Icount_cuda_kernel(LOOP_BODY loop_body,
                                           Index_type length,
                                           Index_type icount)
 {
-  Index_type ii = blockDim.x * blockIdx.x + threadIdx.x;
+
+  auto body = loop_body;
```

Comment:
auto body = loop_body triggers the copy constructor for all captured reduction variables, this is required to sequence the 3 phase construct properly. In the older code, a copy was being triggered by the reduction's operator return, which was returned by value. In the new scheme the operator signature is

__device__ ReduceSum<cuda_reduce<BLOCK_SIZE>, T> const &operator+=(                                                                                                                                                                                              T val) const   

which passes back a reference and does not invoke a copy.